app/models/patient.rb

```diff
@@ -438,6 +438,20 @@ def self.responder_for_email(email)
       .where('responder_id = id')
   end
 
+  def self.dependent_ids_for_patients(patient_ids)
+    return nil unless patient_ids.respond_to?(:each)
+
+    dependent_ids = []
+    Patient.find(patient_ids).each do |patient|
+      next if patient.self_reporter_or_proxy?
+
+      dependent_ids << patient.dependents.pluck('id')
+    end
+    dependent_ids.reject! { |value| value.nil? || value.empty? }
```

Comment:
```suggestion
    dependent_ids = dependent_ids.reject! { |value| value.nil? || value.empty? }
```

Reply:
or just simply not include line 452 that way the result of line 450 gets returned?

Reply:
Your suggestion won't work because `reject!` returns `nil` if nothing changed so the result of the function will be `nil` so we always need to make sure to return `dependent_ids`

Reply:
This (the `!`) is a ruby-ish thing for "modify the object instance itself instead of returning a new copy with the changes made".

Reply:
ah I was looking at the documentation for `reject` and not `reject!`